### ISSUE TITLE
Please make `psycopg2` test dependency optional and/or support psycopg-3

### ISSUE THREAD

**Issue by mgorny** [vutran1710/PyrateLimiter#163]
The `psycopg2` has been superseded a while ago by [psycopg](https://pypi.org/project/psycopg/) >= 3. However, the most recent release make it impossible to run the test suite without `psycopg2` installed:

```pytb
tests/conftest.py:20: in <module>
    from psycopg2.pool import ThreadedConnectionPool
E   ModuleNotFoundError: No module named 'psycopg2'
```

Could you please make the dependency optional, so we could deselect the tests requiring it? Or at least move to the modern `psycopg` package?

**Comment by vutran1710** [comment]
Oh my bad, tks for letting me know

**Comment by vutran1710** [comment]
@mgorny can you review? #164

**Comment by vutran1710** [comment]
Resolved in #164  and #165
